tracer/test/Datadog.Trace.Tests/SpanTests.cs

```diff
@@ -360,5 +360,45 @@ public void SetTag_Double()
             span.GetTag(stringKey).Should().Be(stringValue);
             span.GetTag(numericKey).Should().BeNull();
         }
+
+        [Fact]
+        public void ServiceOverride_WhenSet_HasBaseService()
+        {
+            var origName = "MyServiceA";
+            var span = _tracer.StartSpan(nameof(SetTag_Double), serviceName: origName);
+            span.ServiceName = "MyServiceB";
+
+            span.GetTag(Tags.BaseService).Should().Be(origName);
```

Comment:
nit: maybe we should assert we correctly assert we changed the service. We obviously can, but just for completeness 🙂 

```suggestion
            span.ServiceName.Should().Be("MyServiceB");
            span.GetTag(Tags.BaseService).Should().Be(origName);
```